agent/src/com/jonnyzzz/teamcity/virtual/run/RelativePaths.java

```diff
@@ -26,13 +27,23 @@
  */
 public class RelativePaths {
   @NotNull
-  public static String resolveRelativePath(@NotNull final File baseDir, @NotNull final File path) {
-    String result = FileUtil.getRelativePath(baseDir, path);
+  public static String resolveRelativePath(@NotNull final File baseDir, @NotNull final File path, boolean isWindows, boolean convertToLinux) {
+    String result = isWindows
+            ? FileUtil.getRelativePath(baseDir.getAbsolutePath(), path.getAbsolutePath(), '\\')
+            : FileUtil.getRelativePath(baseDir.getAbsolutePath(), path.getAbsolutePath(), '/');
     if (result == null) return "";
-
-    result = result.replace('\\', '/');
-
-    while (result.endsWith(".") || result.endsWith("/")) result = result.substring(0, result.length() - 1);
+    if(convertToLinux)
+      result = result.replace('\\', '/');
+    if(isWindows) {
+      while (result.endsWith(".") || result.endsWith("\\")) result = result.substring(0, result.length() - 1);
+    } else {
+      while (result.endsWith(".") || result.endsWith("/")) result = result.substring(0, result.length() - 1);
+    }
     return result;
   }
+
+  @NotNull
+  public static String resolveRelativePath(@NotNull final File baseDir, @NotNull final File path) {
+    return resolveRelativePath(baseDir, path, SystemInfo.isWindows, true);
```

Comment:
I mean this one `jetbrains.buildServer.agent.BuildAgentConfiguration#getSystemInfo`. An instance of `BuildAgentConfiguration` is available via Spring autowiring, so you may have an instance from `DockerVM` contructor

Maybe it's too complicated, but I see it a nice idea to check the OS once somewhere on the `DockerContext` instance creation